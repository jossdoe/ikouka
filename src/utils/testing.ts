// This is a proxy-file to allow for global wrappers and modifications
// to the functionality of React Testing Library.
// https://testing-library.com/docs/react-testing-library/setup#custom-render
export * from '@testing-library/react'
