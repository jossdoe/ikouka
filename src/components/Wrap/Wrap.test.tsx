import React from 'react'
import { render, screen } from '../../utils/testing'
import { Wrap, WrapProps } from './index'

const SPACES: WrapProps['space'][] = ['xs', 'sm', 'md', 'lg', 'xl', undefined]

SPACES.forEach((space) => {
  test(`Wrap - ${space}`, () => {
    const { asFragment } = render(
      <Wrap space={space}>
        <button />
        <button />
        <button />
      </Wrap>
    )

    expect(asFragment()).toMatchSnapshot()

    expect(screen.getAllByRole('button')).toHaveLength(3)
  })
})
