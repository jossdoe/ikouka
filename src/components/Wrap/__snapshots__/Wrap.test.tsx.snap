// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrap - lg 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap gap-6 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;

exports[`Wrap - md 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap gap-4 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;

exports[`Wrap - sm 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap gap-2 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;

exports[`Wrap - undefined 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap gap-4 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;

exports[`Wrap - xl 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap gap-10 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;

exports[`Wrap - xs 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap gap-0.5 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;
